### DISCUSSION TITLE
Filter some results and prioritize others

### DISCUSSION THREAD

**Discussion by theonlyalm** [searxng/searxng#3217] [Q&A]
I'm asking about how to filter some results out like if the search query has the word "Texas" it filter the results of BBC.com...and how to make the results of CNN.com be shown at the first top results

I know there's ( hostname_replace) plugin put I don't think it fits my case as I want to filter some domains and give priority to other ones ONLY IF THE SEARCH QUERY HAS A SPECIFIC WORD like "texas" for example

**Comment by virtadpt** [comment]
That's the sort of thing that post-processing is best suited for, say, by software that runs queries using the SearxNG API or search result feeds.

**Reply by theonlyalm** [reply]
So what is the file that is responsible for showing results? so I can rearrange the results based on my desired domains

And in this particular file, how can I know what was the search query so if the query contains the word "texas" I rearrange the results and if it doesn't I let them as they're
